deploy/deployctl/subcommands/dataproc_cluster.py

```diff
@@ -24,6 +25,19 @@ def start_cluster(name: str, cluster_args: typing.List[str]) -> None:
         raise RuntimeError("project configuration is required")
 
     with open(os.path.join(DATA_PIPELINE_DIRECTORY, "requirements.txt")) as requirements_file:
+        requirements_hail_version = next(
+            (line.strip().split("==")[1] for line in requirements_file if line.strip().startswith("hail==")), None
+        )
+        local_hail_version = importlib.metadata.version("hail")
```

Comment:
It looks like `.version()` raises an error (PackageNotFoundError) if hail isn't installed (rather than returning None):

```
>>> local_hail_version = importlib.metadata.version("hail")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/sjahl/.asdf/installs/python/3.9.16/lib/python3.9/importlib/metadata.py", line 569, in version
    return distribution(distribution_name).version
  File "/Users/sjahl/.asdf/installs/python/3.9.16/lib/python3.9/importlib/metadata.py", line 542, in distribution
    return Distribution.from_name(distribution_name)
  File "/Users/sjahl/.asdf/installs/python/3.9.16/lib/python3.9/importlib/metadata.py", line 196, in from_name
    raise PackageNotFoundError(name)
importlib.metadata.PackageNotFoundError: hail
```

You may want to catch that and handle it, since I think we'll just crash there instead of making it to the `if not` check on `local_hail_version`